python/variable.cpp

```diff
@@ -795,4 +795,17 @@ void init_variable(py::module &m) {
         py::arg("x"), py::arg("nan") = std::optional<VariableConstView>(),
         py::arg("posinf") = std::optional<VariableConstView>(),
         py::arg("neginf") = std::optional<VariableConstView>(), py::arg("out"));
+
+  m.def("is_events",
+        [](const VariableConstView &self) { return is_events(self); },
+        R"(Return true if the variable contains event data.)");
+  m.def(
+      "is_events",
```

Comment:
Not tested yet

Reply:
The fate of this is unclear and it is only used for internals in visualization (thus not included in the docs). The changes here include the move from "implicit" data for events to one with explicit weights that may be broadcast. It is not clear yet if this is the right approach, and how it should be handled, e.g., whether it should "present" itself as event data despite being dense, as done in the implementation here. This will also tie into upcoming change with unaligned data. What should `is_events` return for a data array with unaligned event data? The answer is not clear, and I wanted to avoid refactoring tests multiple times. I am adding a tasks to #998.